internals-js/src/federation.ts

```diff
@@ -1736,6 +1737,100 @@ export class FederationBlueprint extends SchemaBlueprint {
       }
     }
 
```

Comment:
I know I didn't necessarily follow this for context, but we should probably make this a separate function.

Reply:
I kept this block, but moved the individual validations into separate functions.